lib/internal/repl.js

```diff
@@ -55,10 +55,20 @@ function createRepl(env, opts, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
-    return setupHistory(repl, env.NODE_REPL_HISTORY,
-                        env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal) {
+    // Make sure we trim the history path. We don't want to create
+    // a file name " ".
+    let historyPath = env.NODE_REPL_HISTORY;
+    if (historyPath) {
+      historyPath = historyPath.trim();
+    }
+
+    if (historyPath !== '') {
+      return setupHistory(repl, env.NODE_REPL_HISTORY,
```

Comment:
Oh wow. Not what I meant to do. Will fix shortly. I was meaning to pass the trimmed historyPath to setupHistory